File: src/main/java/io/github/slimefunguguproject/bump/utils/JsonUtils.java
package io.github.slimefunguguproject.bump.utils;

import javax.annotation.Nonnull;

import com.google.gson.JsonElement;
import com.google.gson.JsonParser;
import com.google.gson.stream.JsonReader;

import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion;
import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;

import lombok.experimental.UtilityClass;

/**
 * Utility methods for json parsing.
 *
 * @author ybw0014
 */
@UtilityClass
public final class JsonUtils {
    /**
     * This method parses JSON from the {@link JsonReader}.
     *
     * @param reader The {@link JsonReader} to be read JSON from.
     *
     * @return The root {@link JsonElement}
     */
    public static JsonElement parseReader(@Nonnull JsonReader reader) {
        if (Slimefun.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_18)) {
            return JsonParser.parseReader(reader);
        } else {
            return new JsonParser().parse(reader);
        }
    }
}

File: src/main/kotlin/io/github/slimefunguguproject/bump/utils/JsonUtils.kt
@file:Suppress("deprecation")

package io.github.slimefunguguproject.bump.utils

import com.google.gson.JsonElement
import com.google.gson.JsonParser
import com.google.gson.stream.JsonReader
import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion
import io.github.thebusybiscuit.slimefun4.implementation.Slimefun

object JsonUtils {
    /**
     * This method parses JSON from the [JsonReader].
     *
     * @param reader The [JsonReader] to be read JSON from.
     *
     * @return The root [JsonElement]
     */
    fun parseReader(reader: JsonReader): JsonElement {
        return if (Slimefun.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_18)) {
            JsonParser.parseReader(reader)
        } else {
            JsonParser().parse(reader)
        }
    }
}
